app/services/forest_liana/filter_parser.rb

```diff
@@ -0,0 +1,245 @@
+    end
+
+    def parse_condition(condition)
+      operator = condition['operator']
+      value = condition['value']
+      field = condition['field']
+
+      if @operator_date_interval_parser.is_date_interval_operator(operator)
+        condition = @operator_date_interval_parser.get_interval_date_filter(operator, value)
+        return "#{parse_field_name(field)} #{condition}"
+      end
+
+      if is_belongs_to(field)
+        association = field.partition(':').first.to_sym
+        association_field = field.partition(':').last
+
+        unless @resource.reflect_on_association(association)
+          raise ForestLiana::Errors::HTTP422Error.new("Association '#{association}' not found")
+        end
+
+        current_resource = @resource.reflect_on_association(association).klass
+      else
+        association_field = field
+        current_resource = @resource
+      end
+
+      # NOTICE: Set the integer value instead of a string if "enum" type
+      # NOTICE: Rails 3 do not have a defined_enums method
+      if current_resource.respond_to?(:defined_enums) && current_resource.defined_enums.has_key?(association_field)
+        value = current_resource.defined_enums[association_field][value]
+      end
+
+      if Rails::VERSION::MAJOR >= 5
+        ActiveRecord::Base.sanitize_sql([
+          "#{parse_field_name(field)} #{parse_operator(operator)}?",
+          parse_value(operator, value)
+        ])
+      else
+        "#{parse_field_name(field)} #{parse_operator(operator)} #{ActiveRecord::Base.sanitize(parse_value(operator, value))}"
+      end
+    end
+
+    def parse_aggregator_operator(aggregator_operator)
+      unless AGGREGATOR_OPERATOR.include?(aggregator_operator)
+        raise ForestLiana::Errors::HTTP422Error.new("Unknown provided operator '#{aggregator_operator}'")
+      end
+
+      aggregator_operator.upcase
+    end
+
+    def parse_operator(operator)
+      case operator
+      when 'not'
+        'NOT'
+      when 'greater_than', 'after'
+        '>'
+      when 'less_than', 'before'
+        '<'
+      when 'contains', 'starts_with', 'ends_with'
+        'LIKE'
+      when 'not_contains'
+        'NOT_LIKE'
+      when 'not_equal'
+        '!='
+      when 'present'
+        'IS NOT NULL'
+      when 'equal'
+        '='
+      when 'blank'
+        'IS NULL'
+      else
+        raise ForestLiana::Errors::HTTP422Error.new("Unknown provided operator '#{operator}'")
+      end
+    end
+
+    def parse_value(operator, value)
+      case operator
+      when 'not'
+        value
+      when'greater_than', 'less_than', 'not_equal', 'equal', 'before', 'after'
+        "#{value}"
+      when 'contains', 'not_contains'
+        "%#{value}%"
+      when 'starts_with'
+        "#{value}%"
+      when 'ends_with'
+        "%#{value}"
+      when 'present', 'blank'
+      else
+        raise ForestLiana::Errors::HTTP422Error.new("Unknown provided operator '#{operator}'")
```

Comment:
[Code Style] Error message duplicated in the method above.
If the logic is to have the exact same message between them, it should be factorised.
If a developper change the sentence, it should be done once not twice.

Reply:
Same sentence at lines 80, 107, 125